python/SAGEMAKER_NOTEBOOK_KMS_CONFIGURED/SAGEMAKER_NOTEBOOK_KMS_CONFIGURED.py

```diff
@@ -107,7 +108,7 @@ def evaluate_compliance(event, configuration_item, valid_rule_parameters):
         elif not valid_rule_parameters:
             evaluations.append(build_evaluation(notebook_instance['NotebookInstanceName'], 'COMPLIANT', event))
         #SCENARIO 5: KMS key specified for Amazon SageMaker Notebook Instance matches keyArn in rule parameter.
-        elif notebook_instance_description['KmsKeyId'] in valid_rule_parameters:
+        elif reduce((lambda x, y: x or y), [notebook_instance_description['KmsKeyId'] in keyId for keyId in valid_rule_parameters]):

```

Comment:
Please explain the value of doing this. "valid_rule_parameters" is a list of key and the notebook_instance_description['KmsKeyId'] is a single key.

Reply:
I am assuming that the KMS key ID is given at the `notebook_instance_description['KmsKeyId']` and this is a substring of the ARN that's passed into the rule parameters. The reduce function will be given an array of boolean values (ex, `[False, False, True]`) and it'll return True if _one_ item in the array is true. This way, we can see if an KMS key ID exists anywhere in the list of KMS ARNs.

Reply:
valid_rule_parameters is a list of all the KmsKeyARN that are authorized. If the instance key matches any of those key, it is considered compliant. The reduce is not necessary here.